### PR TITLE
feature/de-3-item-hidden-but-harvestable

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|-----:|-----:|-------:|
| ETA                  |  20.8  |    0 |     0 |      0 |        0 |
| Developing      |  0  |    12 |    0 |      0 |         0 |
| Review             |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -   |   -     |        0 |
## Problem description
Some Items cannot be searchable but must be harvestable by OAI-PMH. The Items which has metadata "local.hidden" = "yes" should not be indexed to the Solr and cannot be searchable but must be harvestable by OAI-PMH.

- [x] Create a new local-type "local.hidden"
- [x] Create a test to check if DSpace has this new local-type. Class `LocalMetadataTest`
probably here https://github.com/ufal/clarin-dspace/blob/clarin/dspace/config/input-forms.xml
- [x] Add a new input field to the `submission-forms.xml` like in the Clarin-DSpace
- [x] Update indexing of the Item after creating - if the Item has "local.hidden" do not index it

#### How to test:
- [x] Admin should see a new input field - is hidden
- [x] User cannot see a new input field - is hidden
- [x] If the Item is created with metadata "local.hidden" it cannot be searchable - write the Item name to the search bar and it won't be found.
- [x] The Item metadata should be seen in the OAI-PMH page. (Do not forget to update OAI indexes)

## Analysis
(Write here, if there is needed describe some specific problem. Erase it, when it is not needed.)
## Problems
(Write here, if some unexpected problems occur during solving issues. Erase it, when it is not needed.)

### CODE REVIEW

**Comment by MajoBerger** [comment]
@milanmajchrak 
Can this

image

or this

image


be used?

**Comment by MajoBerger** [comment]
@milanmajchrak [AFAIK](https://www.google.com/search?q=afaik+meaning&oq=AFAIK+meaning&aqs=chrome.0.0i512l9j0i390.1848j0j7&sourceid=chrome&ie=UTF-8), OAI-PMH takes items from solr, so how can it be in OAIPMH and not in SOLR?

**Comment by MajoBerger** [comment]
Item when *withdrawn*:
image
not seen in rest thru items, seen with handle

database (withdrawn = true):
image

nothing special seen in metadata

item when *private*:
image

seen in rest

nothing special in metadata

database:
image

**Comment by milanmajchrak** [comment]
> 

@MajoBerger check this thread: https://dspace-org.slack.com/archives/C03JD6V3UUA/p1659606594314449

**Comment by MajoBerger** [comment]
`local.hidden = hidden` makes item deleted - cannot be searched and no metadata seen in OAI-PMH. However, that means there isn't uploaded cmdi file either.

**Comment by MajoBerger** [comment]
This appears problematic. Don't yet know how clarin did it, awainting answer from @shark in Slack. 

Problem is, that all places use the same method (#isDeleted()), but we need OAIPMH to use different one to REST, frontend, search and such. OAI uses library (xoai) which I don't want to modify. Will wait for @shark a bit more.

**Comment by MajoBerger** [comment]
Field `local.hidden` has been renamed to `local.hiddenButHarvestable` to better describe what it is used for.

File: dspace-oai/src/main/java/org/dspace/utils/LicenseUtil.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    private static final Set<String> unrestricted = Collections.unmodifiableSet(new HashSet<String>());
+
+    private static final Map<String, String> _uri2metashare = uri2metashare();
+    private static final Map<String, String> _uri2restrictions = uri2restrictions();
+
+    private static Map<String, String> uri2metashare() {
+        HashMap<String, String> map = new HashMap<String, String>();
+        for (String def : _uri2metashareDefs) {
+            String[] defn = def.split(",", 2);
+            map.put(defn[0], defn[1]);
```
Comment (milanmajchrak) [inline dspace-oai/src/main/java/org/dspace/utils/LicenseUtil.java:140]: if defn has 1 value

File: dspace-oai/src/main/java/org/dspace/utils/LicenseUtil.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        }
+        return Collections.unmodifiableMap(map);
+    }
+
+    private static Map<String, String> uri2restrictions() {
+        HashMap<String, String> map = new HashMap<String, String>();
+        for (String def : _restrictionDefs) {
+            String[] defn = def.split("©", 2);
+            map.put(defn[0], defn[1]);
```
Comment (milanmajchrak) [inline dspace-oai/src/main/java/org/dspace/utils/LicenseUtil.java:149]: if defn has 1 value

**Pull request merged**